src/go/k8s/pkg/resources/configmap.go

```diff
@@ -53,27 +59,56 @@ type ConfigMapResource struct {
 	scheme		*runtime.Scheme
 	pandaCluster	*redpandav1alpha1.Cluster
 
-	svc	*ServiceResource
-	logger	logr.Logger
+	svc		*HeadlessServiceResource
+	logger		logr.Logger
+	nodePort	int32
 }
 
 // NewConfigMap creates ConfigMapResource
 func NewConfigMap(
 	client k8sclient.Client,
 	pandaCluster *redpandav1alpha1.Cluster,
 	scheme *runtime.Scheme,
-	svc *ServiceResource,
+	svc *HeadlessServiceResource,
 	logger logr.Logger,
 ) *ConfigMapResource {
 	return &ConfigMapResource{
-		client, scheme, pandaCluster, svc, logger.WithValues("Kind", configMapKind()),
+		client,
+		scheme,
+		pandaCluster,
+		svc,
+		logger.WithValues("Kind", configMapKind()),
+		0,
 	}
 }
 
 // Ensure will manage kubernetes v1.ConfigMap for redpanda.vectorized.io CR
-//nolint:dupl // we expect this to not be duplicated when more logic is added
 func (r *ConfigMapResource) Ensure(ctx context.Context) error {
 	var cfgm corev1.ConfigMap
+	sb := strings.Builder{}
```

Comment:
should we go for something more specific like ips at least? or ipsSb?

Reply:
Ok will change to IPsList